### ISSUE TITLE
Introducing Gitomator

### ISSUE THREAD

**Issue by joeyfreund** [education/teachers#49]
Hi there,

My name is Joey Freund, I am a lecturer at the University of Toronto, and I have been using GitHub in our advanced Software Engineering course for almost two years now.

I wanted to introduce Gitomator, https://gitomator.github.io/ 

 > **Important:** Gitomator is in pre-alpha stage. That means that some features are not yet implemented, there aren't enough tests and the docs are constantly a work-in-progress. If that scares you, I totally understand, please check back in a coupe of weeks.

#### What is Gitomator?

Workflow automation tools for software educators.

Gitomator comes with command-line tools that let you conveniently use GitHub organizations in software engineering courses. For example: automate the process of creating repos and/or teams, managing team memberships and access-permissions, merging pull-requests, enabling/disabling CI, auto-marking and more.

#### Why did I build it?

In the course I was teaching, I wanted to implement a specific workflow:

 * Individual assignments are distributed as private repos (one repo per student).
 * Student get read-only permission to their repo, do their work in a fork, and submit their solution via pull-request.
 * Students benefit from immediate feedback on their work, via Travis CI.
 * Important feature: Easy to push code updates after an assignment was released to the students.

I hacked together a prototype to help me automate the workflow above (and a few additional workflow we ended using). My experience was great, and after using the prototype for two semesters, I decided it's time to build a less-prototypical, more generally-purpose version of the tool.

#### Why am I sharing it with you?

Because ...

 1. I want others to use it
 2. I want others to join the project - Gitomator is an open-source project, currently in pre-alpha stage, and I would really love to see a contributor-friendly community growing around it.

Feel free to ask questions, leave feedback, request features, suggest improvements, etc.

**Comment by thomasjbradley** [comment]
How has your success been with Travis CI in class?

I was using it for a little while but the students were getting extremely frustrated with waiting for their tests to run. Now, it's probably that the tests were complex, and the students aren't computer science majors so they couldn't run them locally. (They're graphic design students learning basic web development.)

**Comment by joeyfreund** [comment]
Our experience was great.

Our students did most of their development work (including running the tests) on their local machine, and used Travis as a "verification tool" for their code submission.

 * Since we were teaching advanced software engineering, there is the obvious benefit of providing our students hands-on experience with industry standard tools. Some of our students decided to use Travis CI for their team projects as well.

 * Next, Travis CI was a great time-saver:
    * Our individual coding assignments were distributed as Java Maven projects. Travis CI helped catching various compilation errors, before it was too late. Since our class was fairly large (~170 students), preventing technical issues in advance meant a lot of time saved.
    * In the few cases where our auto-marker and Travis CI gave different results, having the build log available on-line, made my debugging a whole lot easier (usually took me less than a minute to figure out what happened).

 * Then, there is the point about Travis CI giving students almost-immediate feedback (as soon as a students got a green light from Travis, they knew they were done their assignment). By taking away the uncertainty and the wait-time, we hoped to make our assignments less stressful.

**Comment by thomasjbradley** [comment]
Yeah, it does make lots of sense for computer engineering students. I ended up moving away from Travis CI to a local app the students could easily drag-and-drop their code into for testing and auto-marking.
